servers/rendering/renderer_rd/environment/sky.cpp

```diff
@@ -630,14 +630,39 @@ bool SkyRD::Sky::set_radiance_size(int p_radiance_size) {
 	return true;
 }
 
+RS::SkyMode SkyRD::Sky::get_effective_mode() const {
+	if (mode != RS::SKY_MODE_AUTOMATIC) {
+		return mode;
+	}
+
+	SkyMaterialData *material_data = static_cast<SkyMaterialData *>(RendererRD::MaterialStorage::get_singleton()->material_get_data(material, RendererRD::MaterialStorage::SHADER_TYPE_SKY));
```

Comment:
This will fail when using the compatibility renderer since it does not use RendererRD